### PR TITLE
feat(validator_node): add get_sidechain_block p2p rpc method

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---

- Add get_sidechain_blocks to validator node p2p rpc
- Use fixed 32-byte array for hashes on dan layer
- Implement FromStr and fallible integer conversions for TemplateId
- Remove asset_public_key from Instruction
- Add necessary chain db calls
- Add client call for get_sidechain_blocks
- Adds mocks and basic tests for get_sidechain_blocks validator node rpc

Motivation and Context
---
Allow client to fetch sidechain instruction state from vn

How Has This Been Tested?
---
Manually

### CODE REVIEW

File: dan_layer/core/src/types.rs [hunk 1]
```diff
@@ -26,6 +26,8 @@ use tari_crypto::{
     tari_utilities::{ByteArray, ByteArrayError},
 };
 
+pub type Hash = [u8; 32];
```
Comment (stringhandler) [inline dan_layer/core/src/types.rs]: Not used?

**Review by stringhandler — DISMISSED** [review]
Nice. Thanks for cleaning up some of the dirty bits I left

File: dan_layer/core/src/types.rs (hunk 1 above)
Comment (sdbondi) [reply dan_layer/core/src/types.rs]: Don't think it's used, will remove in next PR (adding to client and some tests) if this is merged tomorrow

Comment (sdbondi) [reply dan_layer/core/src/types.rs]: Removed in this PR since there was a conflict

**Review by delta1 — APPROVED** [review]

**Pull request merged**